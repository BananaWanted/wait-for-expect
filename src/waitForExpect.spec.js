/* eslint-env jest */
const waitForExpect = require("./index");

let numberToChange = 10;

test("it waits for the number to change", async () => {
  setTimeout(() => {
    numberToChange = 100;
  }, 600);
  await waitForExpect(() => {
    expect(numberToChange).toEqual(100);
  });
});
