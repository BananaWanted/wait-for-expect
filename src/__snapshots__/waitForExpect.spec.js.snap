// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it fails properly when it times out without expectation passing 1`] = `
[Error: [2mexpect([22m[31mreceived[39m[2m).toEqual([22m[32mexpected[39m[2m)[22m

Expected value to equal:
  [32m2000[39m
Received:
  [31m200[39m]
`;

exports[`it fails properly with jest error message when it times out without expectation passing 1`] = `
[Error: [2mexpect([22m[31mreceived[39m[2m).toEqual([22m[32mexpected[39m[2m)[22m

Expected value to equal:
  [32m2000[39m
Received:
  [31m200[39m]
`;

exports[`it fails when the change didn't happen fast enough, based on the waitForExpect timeout 1`] = `
[Error: [2mexpect([22m[31mreceived[39m[2m).toEqual([22m[32mexpected[39m[2m)[22m

Expected value to equal:
  [32m3000[39m
Received:
  [31m300[39m]
`;
